auditeval/test.go

```diff
@@ -44,85 +51,52 @@ type compare struct {
 
 type testOutput struct {
 	TestResult   bool
-	ActualResult string
+	ActualResult []Attributes
 }
 
+type Attribute struct {
+	Name  string
+	Value interface{}
+}
+
+func (a *Attribute) Print() {
+	fmt.Printf("\t%s: %v", a.Name, a.Value)
+}
+
+type Attributes []Attribute
+
 func (t *testItem) execute(s string) *testOutput {
-	result := &testOutput{TestResult: false, ActualResult: ""}
+	result := &testOutput{TestResult: true, ActualResult: []Attributes{}}
 
 	s = strings.TrimRight(s, " \n")
-	if t.Set {
-		if t.Compare.Op != "" {
-			flagVal := getFlagValue(s, t.Flag)
-			result.ActualResult = strings.ToLower(flagVal)
-
-			switch t.Compare.Op {
-			case "eq":
-				value := strings.ToLower(flagVal)
-				// Do case insensitive comparaison for booleans ...
-				if value == "false" || value == "true" {
-					result.TestResult = value == t.Compare.Value
-				} else {
-					result.TestResult = flagVal == t.Compare.Value
-				}
-
-			case "noteq":
-				value := strings.ToLower(flagVal)
-				// Do case insensitive comparaison for booleans ...
-				if value == "false" || value == "true" {
-					result.TestResult = !(value == t.Compare.Value)
-				} else {
-					result.TestResult = !(flagVal == t.Compare.Value)
-				}
-
-			case "gt":
-				a, b, err := toNumeric(flagVal, t.Compare.Value)
-				if err == nil {
-					result.TestResult = a > b
-				}
-
-			case "gte":
-				a, b, err := toNumeric(flagVal, t.Compare.Value)
-				if err == nil {
-					result.TestResult = a >= b
-				}
-
-			case "lt":
-				a, b, err := toNumeric(flagVal, t.Compare.Value)
-				if err == nil {
-					result.TestResult = a < b
-				}
-
-			case "lte":
-				a, b, err := toNumeric(flagVal, t.Compare.Value)
-				if err == nil {
-					result.TestResult = a <= b
-				}
-
-			case "has":
-				result.TestResult = strings.Contains(flagVal, t.Compare.Value)
-
-			case "nothave":
-				result.TestResult = !strings.Contains(flagVal, t.Compare.Value)
+	flagReg := regexp.MustCompile(t.Flag)
+
+	// If the test should run on multipul values - Meaning if the flag occures more than once (Containers, Images, etc)
+	if len(flagReg.FindAllStringIndex(s, -1)) > 1 {
+		values := strings.Split(s, "\n")
+		testResult := true
+
+		for _, v := range values {
+			testResult = t.evalTestResult(v)
+
+			if !testResult {
+				result.TestResult = false
+				result.ActualResult = smartAppend(result.ActualResult, parseActualResult(v, t.Flag))
```

Comment:
OK, After we will agree on the implementation, I'll add unit tests.

An example for this will be a test that returns a list of containers that fails the test. a test has multiple checks than the list of the failed containers will have some of the same containers multiple times.